backend/tests/test_typography.py

```
@@ -0,0 +1,73 @@
+            patch("app.Template", side_effect=capture_template),
+            self.assertRaises(TemplateCaptured),
+        ):
+            asyncio.run(
+                generate_handwriting_impl(
+                    "http://localhost/",
+                    params,
+                    font_file=b"test-font",
+                )
+            )
+
+        expected_end_chars = "，。》？；：’”】｝、！％）,.>?;:]}!%)′″℃℉"
```

Ruff (0.15.21)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)